### PR TITLE
Collection-returning REST API and string id

### PR DESCRIPTION
**Pull request by frsyuki**
This PR rebased #333 and #323 from master and added changes of digdag-ui.

### CODE REVIEW

**Comment by frsyuki** [comment]
@dzannotti could you review changes of digdag-ui? I think they're ok because `npm run flow` passed. 

@danielnorberg could you review backend changes? There're 4 kinds of changes:

* digdag-client and digdag-cli
  * Most of changes are here. They use `io.digdag.client.api.Id` instead of `long` or `int` to represent ids. I created the wrapper class because javac can do more type checking.
* digdag-server
  * Response body uses Id instead of long/int for ID but query parameters keep using int/long. RESTeasy (JAX-RS) converts request params to int/long automatically.
* digdag-tests
  * All simple replacements. For example,
     * `Long.toString(attemptId)` -> `attemptId.toString()` (where attemptId is changed from long to Id)
     * `List<RestSchedule> scheds = client.getSchedules()` -> `List<RestSchedule> scheds = client.getSchedules().getSchedules()`
* digdag-core
  * Added `io.digdag.core.database.IdName` class because DatabaseProjectStoreManager was reusing api.IdName.

I'm thinking that next version should be `0.9.0` together with other breaking changes: Operator API (#344), and secret's - to _ change (#399).
We also have some fancy changes to convince users to upgrade: digdag-ui integration (#401), emr operators (#397), and redshift operators (#292).

File: digdag-ui/console.jsx [hunk 1]
```diff
@@ -478,9 +478,9 @@ class ScheduleListView extends React.Component {
 
   fetchSchedule () {
     this.setState({loading: true})
-    model().fetchProjectWorkflowSchedule(this.props.projectId, this.props.workflowName).then(schedules => {
+    model().fetchProjectWorkflowSchedule(this.props.projectId, this.props.workflowName).then(scheduleCollection => {
```
Comment (dzannotti) [inline digdag-ui/console.jsx]: I think it's worth destructuring here so we can use the short-hand for object

```
then(({ schedules }) => {
  this.setState({
     schedules,
     loading: false		          loading: false
   })
})
```

File: digdag-ui/console.jsx [hunk 2]
```diff
@@ -593,8 +593,8 @@ class SessionsView extends React.Component {
   };
 
   componentDidMount () {
-    model().fetchSessions().then(sessions => {
-      this.setState({sessions})
+    model().fetchSessions().then(sessionCollection => {
```
Comment (dzannotti) [inline digdag-ui/console.jsx]: same as above

File: digdag-ui/console.jsx [hunk 3]
```diff
@@ -571,8 +571,8 @@ class ProjectsView extends React.Component {
   };
 
   componentDidMount () {
-    model().fetchProjects().then(projects => {
-      this.setState({projects})
+    model().fetchProjects().then(projectCollection => {
```
Comment (dzannotti) [inline digdag-ui/console.jsx]: same as above

File: digdag-ui/console.jsx (hunk 1 above)
Comment (frsyuki) [inline digdag-ui/console.jsx]: that's an interesting syntax. I didn't know that. Updating the code...

File: digdag-ui/console.jsx [hunk 4]
```diff
@@ -1337,9 +1337,9 @@ class AttemptLogsView extends React.Component {
   }
 
   fetchLogs () {
-    model().fetchAttemptLogFileHandles(this.props.attemptId).then(files => {
+    model().fetchAttemptLogFileHandles(this.props.attemptId).then(fileCollection => {
       if (!this.ignoreLastFetch) {
-        const sortedFiles = _.sortBy(files, 'fileTime')
+        const sortedFiles = _.sortBy(fileCollection.files, 'fileTime')
```
Comment (dzannotti) [inline digdag-ui/console.jsx]: I'd use destructoring here too

File: digdag-ui/console.jsx (hunk 1 above)
Comment (frsyuki) [inline digdag-ui/console.jsx]: updated 👍

**Comment by dzannotti** [comment]
Left a couple of minor comments, lgtm!

File: digdag-ui/console.jsx (hunk 3 above)
Comment (frsyuki) [inline digdag-ui/console.jsx]: 👍  updated

File: digdag-ui/console.jsx (hunk 4 above)
Comment (frsyuki) [inline digdag-ui/console.jsx]: 👍  updated

File: digdag-ui/console.jsx [hunk 5]
```diff
@@ -1251,9 +1251,9 @@ class AttemptTasksView extends React.Component {
   }
 
   fetchTasks () {
-    model().fetchAttemptTasks(this.props.attemptId).then(tasks => {
+    model().fetchAttemptTasks(this.props.attemptId).then(taskCollection => {
       if (!this.ignoreLastFetch) {
-        this.setState({tasks})
+        this.setState({tasks: taskCollection.tasks})
```
Comment (dzannotti) [inline digdag-ui/console.jsx]: nit: destructuring here too

File: digdag-ui/console.jsx [hunk 6]
```diff
@@ -1635,14 +1635,14 @@ class SessionPage extends React.Component {
...
 
   fetchSession () {
-    model().fetchSession(parseInt(this.props.params.sessionId)).then(session => {
+    model().fetchSession(this.props.params.sessionId).then(session => {
       if (!this.ignoreLastFetch) {
         this.setState({session})
       }
     })
-    model().fetchSessionAttempts(parseInt(this.props.params.sessionId)).then(attempts => {
+    model().fetchSessionAttempts(this.props.params.sessionId).then(attemptCollection => {
```
Comment (dzannotti) [inline digdag-ui/console.jsx]: nit: same

File: digdag-server/src/main/java/io/digdag/server/rs/RestModels.java [hunk 7]
```diff
@@ -153,6 +239,32 @@ public static RestSession session(StoredSessionWithLastAttempt session, String p
...
+                        .collect(Collectors.toList()));
+
+        List<RestSession> collection = sessions.stream()
+                .map(session -> {
+                    try {
+                        return session(session, projs.get(session.getProjectId()).getName());
+                    }
+                    catch (ResourceNotFoundException ex) {
+                        // must not happen
+                        return null;
```
Comment (danielnorberg) [inline digdag-server/src/main/java/io/digdag/server/rs/RestModels.java]: throw here instead of ignoring exception

**Comment by danielnorberg** [comment]
👍

File: digdag-server/src/main/java/io/digdag/server/rs/RestModels.java (hunk 7 above)
Comment (frsyuki) [inline digdag-server/src/main/java/io/digdag/server/rs/RestModels.java]: 👍

**Pull request merged**